### DISCUSSION TITLE
Query - merging RIS files

### DISCUSSION THREAD

**Discussion by Emanuel-1986** [asreview/asreview#1870] [Troubleshooting (report errors in the "Issues" tracker)] [answered]
Hi,

I have downloaded RIS files from Embase(Elsevier), CINAHL (Ebsco), and Epistemonikos. Then, I downloaded the .nbib file from PubMed and converted it to RIS using RefWorks software. 

To check that ASreview reads the RIS files, I used the describe argument and none of the RIS files had any missing titles, and a very small fraction had missing abstracts, but a chunk were unlabelled. When I merged all the 4 RIS files using vstack argument, the number of missing abstracts and titles increased dramatically. This also happened when I merged 2 RIS files, as shown in the table below. 

Can you help me overcome this? Am I doing something wrong with the code when using vstack?

Regards,
Emanuel 

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/Schembri/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/Schembri/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/Schembri/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-GB style='tab-interval:36.0pt;word-wrap:break-word'>

Database /    merged file | Missing abstracts | Missing titles | Unlabelled | Number of records
-- | -- | -- | -- | --
PubMed | 1 | 0 | 1721 | 1721
Embase | 247 | 0 | Null | 7762
CINAHL | 108 | 0 | 1405 | 1405
Epistemonikos | 10 | 0 | Null | 907
Pubmed + Embase | 1968 | 0 | 9483 | 9483
CINAHL + Epistemonikos | 118 | 1405 | 2312 | 2312
All 4   merged | 4280 | 10888 | 11795 | 11795

</body>

</html>

Code:

cd C:\Users\Schembri\Desktop
asreview data describe pubmed.ris
asreview data describe Embase.ris
asreview data describe CINAHL.ris
asreview data describe epistemonikos.ris
asreview data vstack output/merged_RIS_file.ris Pubmed.ris Embase.ris CINAHL.ris epistemonikos.ris
asreview data describe output/merged_RIS_file.ris

**Version information**

- OS: Windows
- Browser: Chrome
- ASReview version: 1.2.1

**Comment by J535D165** [comment]
Thanks for reporting. The issue is related to the somewhat confusing structure of RIS files (which ASReview atm doesn't handle well). 

RIS files can store titles under different "tags" (e.g., TI, T1, T2, T3). For more information, see https://en.wikipedia.org/wiki/RIS_(file_format). ASReview can detect the tag used, but apparently, asreview datatools struggles when files use different tags (for example Embase uses T1 and Epistemonikos uses TI). 

A temporary solution for you would be: 

1. Open each RIS file
2. Search and replace "T1  -" with "TI  -"
3. Repeat for T2 and T3
4. Call vstack again. 

I will open an issue on asreview-datatools to fix this.

**Comment by Emanuel-1986** [comment, marked as answer]
Hi,

Thank you for your prompt response. 

As suggested, I changed:

 "T1  -" to "TI  -"  in Embase, CINAHL, and PubMed RIS files
Abstract from "N2  -" to "AB  -" in Embase RIS file
DOI from "L2  -" to "DO  -" in Embase RIS file

By doing so, the number of missing abstracts has decreased drastically from 4,280 to 366, and the number of missing titles has also reduced from 10,888 to 0, out of a total of 11,795 records when merging the 4 RIS files using vstack.

Reagrds,

Emanuel
